nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/QuerySolr.java

```diff
@@ -0,0 +1,584 @@
+                            try (final RecordSetWriter writer = writerFactory.createWriter(getLogger(), schema, out)) {
+                                writer.write(recordSet);
+                                writer.flush();
+                                mimeType.append(writer.getMimeType());
+                            } catch (SchemaNotFoundException e) {
+                                throw new ProcessException("Could not parse Solr response", e);
+                            }
+                        });
+                        flowFileResponse = session.putAttribute(flowFileResponse, CoreAttributes.MIME_TYPE.key(), mimeType.toString());
+                    }
+
+                    if (processFacetsAndStats) {
+                        if (searchComponents.contains(FacetParams.FACET)) {
+                            FlowFile flowFileFacets = session.create(flowFileResponse);
+                            flowFileFacets = session.write(flowFileFacets, out -> {
+                                try (
+                                        final OutputStreamWriter osw = new OutputStreamWriter(out);
+                                        final JsonWriter writer = new JsonWriter(osw)
+                                ) {
+                                    addFacetsFromSolrResponseToJsonWriter(response, writer);
+                                }
+                            });
+                            flowFileFacets = session.putAttribute(flowFileFacets, CoreAttributes.MIME_TYPE.key(), MIME_TYPE_JSON);
+                            session.transfer(flowFileFacets, FACETS);
+                        }
+
+                        if (searchComponents.contains(StatsParams.STATS)) {
+                            FlowFile flowFileStats = session.create(flowFileResponse);
+                            flowFileStats = session.write(flowFileStats, out -> {
+                                try (
+                                        final OutputStreamWriter osw = new OutputStreamWriter(out);
+                                        final JsonWriter writer = new JsonWriter(osw)
+                                ) {
+                                    addStatsFromSolrResponseToJsonWriter(response, writer);
+                                }
+                            });
+                            flowFileStats = session.putAttribute(flowFileStats, CoreAttributes.MIME_TYPE.key(), MIME_TYPE_JSON);
+                            session.transfer(flowFileStats, STATS);
+                        }
+                        processFacetsAndStats = false;
+                    }
+                }
+
+                if (getEntireResults) {
+                    final Integer totalDocumentsReturned = solrQuery.getStart() + solrQuery.getRows();
+                    if (totalDocumentsReturned < totalNumberOfResults) {
+                        solrQuery.setStart(totalDocumentsReturned);

```

Comment:
I think there should be a sane default limit to how far this goes since it's not using a cursor to get there. As-is, I think it would enable a user to do something like page out tens of thousands of results which AFAIK without a cursor could be pretty bad for Solr's performance.

@bbende @ijokarumawak thoughts on that?

Reply:
@MikeThomsen I could add a property limiting the total amout of results the processor requests. This property could have a default of let's say 10000. If this property is set to 0, there is no limit. The property's description could include a warning with respect to Solr performance issues in the case of deep paging.

Reply:
I think we should set an upper limit of 10000 because this doesn't use the Solr-approved method of deep pagination. In fact I would say that 1,000 to 5,000 should be where the range is with explicit documentation that this is **not** the processor you're looking for if your goal is to hoover up the collection and process it :)

Reply:
FYI, ES has the same issue and the solution in place now is a separate "scroll processor" which would be the equivalent of a "SolrCursorQuery" processor. In general, the use cases here are radically different so for the sake of our sanity and code quality, don't worry about people who want to process everything for the moment.